src/app/modules/project-management/components/create-project/create-project.component.ts

```diff
@@ -1,66 +1,61 @@
-import {
-  Component,
-  Input,
-  OnChanges,
-  Output,
-  EventEmitter
-} from '@angular/core';
+import { Component, Input, OnChanges, OnInit, Output, EventEmitter } from '@angular/core';
 import { FormBuilder, Validators } from '@angular/forms';
+import { Store } from '@ngrx/store';
 import { Project } from '../../../shared/models';
+import * as actions from '../../store/project.actions';
 
 @Component({
   selector: 'app-create-project',
   templateUrl: './create-project.component.html',
-  styleUrls: ['./create-project.component.scss']
+  styleUrls: ['./create-project.component.scss'],
 })
-export class CreateProjectComponent implements OnChanges {
-
+export class CreateProjectComponent implements OnChanges, OnInit {
   projectForm;
   editProjectId;
-  projectStatus = ['', 'Active', 'Inactive'];
+  isEdit = false;
 
   @Input() projectToEdit: Project;
-  @Output() savedProject = new EventEmitter();
   @Output() cancelForm = new EventEmitter();
 
-  constructor(private formBuilder: FormBuilder) {
+  constructor(private formBuilder: FormBuilder, private store: Store<any>) {
     this.projectForm = this.formBuilder.group({
       name: ['', Validators.required],
-      details: ['', Validators.required],
-      status: ['', Validators.required],
-      completed: [false]
+      description: ['', Validators.required],
     });
   }
 
+  ngOnInit(): void {}
+
   ngOnChanges(): void {
     if (this.projectToEdit) {
-      this.editProjectId = this.projectToEdit.id;
-      this.projectForm.setValue({
-        name: this.projectToEdit.name, details: this.projectToEdit.details,
-        status: this.projectToEdit.status, completed: this.projectToEdit.completed
-      });
+      this.projectForm.patchValue(this.projectToEdit);
+      this.isEdit = true;
+    } else {
+      this.projectForm.reset();
+      this.isEdit = false;
     }
   }
 
   get name() {
     return this.projectForm.get('name');
   }
 
-  get details() {
-    return this.projectForm.get('details');
-  }
-
-  get status() {
-    return this.projectForm.get('status');
+  get description() {
+    return this.projectForm.get('description');
   }
 
-  onSubmit(projectData) {
+  onSubmit(formData) {
+    const projectData = { ...this.projectToEdit, ...formData };
+    if (!this.isEdit) {
+      this.store.dispatch(new actions.PostProject(projectData));
+    } else {
+      this.store.dispatch(new actions.PutProject(projectData));
+    }
```

Comment:
In general terms, it's always a good idea to value if(true). In this case, this code should look like:
```
 if (this.isUpdating) {
      this.store.dispatch(new actions.UpdateProject(projectData));
    } else {
      this.store.dispatch(new actions.CreateProject(projectData));
    }
```
I am assuming here the actions have been renamed.